types/src/database_sql.rs

```diff
@@ -1555,6 +1558,74 @@ impl From<BasicWord> for AnnotatedForm {
     }
 }
 
+// Loader to get each chapter id from the database regardless if it's a subchapter or not.
+#[async_trait]
+impl Loader<CollectionChapter> for Database {
+    type Value = Vec<ChapterSingle>;
+    type Error = Arc<sqlx::Error>;
+
+    // string slug can look like "cwkw".chapter1.doc1 But assume user only passes "cwkw"
+
+    async fn load(
+        &self,
+        keys: &[CollectionChapter],
+    ) -> Result<HashMap<CollectionChapter, Self::Value>, Self::Error> {
+        let keys: Vec<_> = keys.iter().map(|k| k.0.clone()).collect();
+        let items = query_file!("queries/collection_chapters.sql", &keys)
+            .fetch_all(&self.client)
+            .await?;
+        Ok(items
+            .into_iter()
+            .map(|chapter| {
+                (
+                    CollectionChapter(chapter.collection_slug),
+                    ChapterSingle {
+                        id: chapter.id,
+                        title: chapter.title,
+                        wordpress_id: chapter.wordpress_id,
+                        index_in_parent: chapter.index_in_parent,
+                        section: if chapter.section == Some("Intro".to_string()) {
+                            CollectionSection::Intro
+                        } else {
+                            CollectionSection::Body
+                        },
+                    },
+                )
+            })
+            .into_group_map())
+    }
+}
+
+#[async_trait]
+impl Loader<CollectionObject> for Database {
+    type Value = Vec<EditedCollection>;
+    type Error = Arc<sqlx::Error>;
+
+    async fn load(
+        &self,
+        keys: &[CollectionObject],
+    ) -> Result<HashMap<CollectionObject, Self::Value>, Self::Error> {
+        let keys: Vec<_> = keys.iter().map(|k| k.0.clone()).collect();
+        let items = query_file!("queries/collection_attributes.sql", &keys)
+            .fetch_all(&self.client)
+            .await?;
+        Ok(items
+            .into_iter()
+            .map(|collection| {
+                (
+                    CollectionObject(collection.slug.clone()),
+                    EditedCollection {
+                        id: collection.id,
+                        title: collection.title,
+                        wordpress_menu_id: collection.wordpress_menu_id,
+                        slug: collection.slug,
+                    },
+                )
+            })
+            .into_group_map())
```

Comment:
Look at other Loader impls that have singular result types, they use a different method than `.into_group_map()`, since you need to assume a single value per key rather than grouping values with the same key.